addons/web/static/src/webclient/navbar/navbar.js

```diff
@@ -64,13 +64,18 @@ export class NavBar extends Component {
             this.render();
         };
 
-        useBus(systrayRegistry, "UPDATE", renderAndAdapt);
-        useBus(this.env.bus, "MENUS:APP-CHANGED", renderAndAdapt);
+        systrayRegistry.on("UPDATE", this, renderAndAdapt);
+        this.env.bus.on("MENUS:APP-CHANGED", this, renderAndAdapt);
         // We don't want to adapt every time we are patched
         // rather, we adapt only when menus or systrays have changed.
         useEffect(() => {this.adapt();}, () => [adaptCounter]);
     }
 
+    willUnmount() {
```

Comment:
use the `onWillUnmount` method (called in `setup`) => the willUnmount component method is no longer called in owl 2, so this will break at some point